--- a/linuxrc.syno.impl
+++ b/linuxrc.syno.impl
@@ -3,6 +3,10 @@

 echo "START /linuxrc.syno.impl"

+set -x
+echo insmod /usr/lib/modules/override.ko; insmod /usr/lib/modules/rp.ko;
+/addons/addons.sh early
+
 . /etc.defaults/rc.subr
 . /usr/syno/share/environments.sh
 . /usr/syno/share/util.sh
@@ -67,6 +71,9 @@
 	# show date for login info
 	date

+	# rmmod redpill
+	/addons/addons.sh jrExit
+
 	UmountRoot

 	exit $1
@@ -176,6 +183,8 @@
 echo "Insert basic USB modules..."
 SYNOLoadModules $USB_MODULES

+/addons/addons.sh modules
+
 # insert Etron USB3.0 drivers

 if [ $KERNEL_VCODE -ge "$(KernelVersionCode "3.10")" ]; then
@@ -224,6 +233,8 @@
 		fi
 fi

+/addons/addons.sh patches
+
 LoadBrmModules

 ####################################################